### ISSUE TITLE
Can't add Syncthing Devices

### ISSUE THREAD

**Issue by slcn-dxd** [KoenDR06/nix-config#1]
I was looking for a way to manage my Syncthing device IDs using sops-nix and found your solution (and this repo consecutively) on Reddit. Unfortunately, I cannot reproduce the desired behavior using your config.

Here is a snippet of my configuration:

```nix
sops = {
  defaultSopsFile = ../secrets.yaml;
  secrets."syncthing/alba" = {
    owner = "danny";
  };
  templates."syncalba".content = ''${config.sops.placeholder."syncthing/alba"}'';
};

services.syncthing = {
  enable = true;
  ...
  settings.devices."alba" = { id = config.sops.templates."syncalba".content; };
  ...  
};
```

After rebuilding my system config, the device is not added in syncthing. Issuing `journalctl | grep syncthing | grep device` results in
```
Sep 10 14:43:31 nycticorax [...]-merge-syncthing-config[...]: "<SOPS:838d94...8d29c:PLACEHOLDER>": device ID invalid: incorrect length
```

So it seems like the placeholder in the syncthing config is not replaced by the ID properly. Do you have an idea how I can fix this?

**UPDATE**: I found the reason why this configuration should not work in the first place: 

From [sops-nix README](https://github.com/Mic92/sops-nix?tab=readme-ov-file#using-secrets-at-evaluation-time):
> It is not possible to use secrets at evaluation time of nix code. This is because sops-nix decrypts secrets only in the activation phase of nixos i.e. in nixos-rebuild switch on the target machine.

So while evaluating the config, the placeholder cannot be replaced. I still wonder how this works on your end. Nevertheless, I wrote a somewhat hacky solution that works by using the `syncthing cli` command to add the devices to Syncthing from the secret files created by sops-nix. Simply add this to your configuration:

```nix
system.userActivationScripts.addSyncthingDevices = {
    text = let 
      hostnamesFormatted = lib.concatStringsSep " " (map (hostname: ''${config.sops.secrets."syncthing/${hostname}-id".path}'') hostnames);
      syncthing = lib.getExe pkgs.syncthing;
    in ''
      #!/bin/bash
      declare -a id_paths=(${hostnamesFormatted})
      for id in "''${id_paths[@]}"
      do
        device_id=$(cat "$id")
	${syncthing} cli config devices add --device-id "$device_id"
      done
    '';
  };
```

This works under the assumption that all devices in your `sops.yaml` are keyed by `<hostname>-id` and that sops-nix makes the respective secrets available for your user (simplay add the option `user = <username>` to each device id secret). Additionally, the `hostnames` variable should be assigned to a list of all hostnames you want to add to Syncthing.

**Comment by KoenDR06** [comment]
Yeah I am aware that it does not work. It's on my todo list of things to fix / do in this repo. I probably need to comment the entire section in machines/common/configuration.nix out.
